paderbox/io/new_subdir.py

```diff
@@ -0,0 +1,341 @@
+import os
+import re
+import time
+import fnmatch
+import datetime
+import urllib.request
+from pathlib import Path
+
+from scipy._lib.decorator import getfullargspec
+
+
+def _removeprefix(self: str, prefix: str) -> str:
+    # Backport from Python 3.9: https://www.python.org/dev/peps/pep-0616/
+    if self.startswith(prefix):
+        return self[len(prefix):]
+    else:
+        return self[:]
+
+
+def _removesuffix(self: str, suffix: str) -> str:
+    # Backport from Python 3.9:  https://www.python.org/dev/peps/pep-0616/
+    # suffix='' should not call self[:-0].
+    if suffix and self.endswith(suffix):
+        return self[:-len(suffix)]
+    else:
+        return self[:]
+
+
+def get_new_subdir(
+        basedir,
+        *,
+        id_naming='index',
+        mkdir=True,
+        prefix=None,
+        suffix=None,
+        consider_mpi=False,
+        dry_run=False,
+):
+    """Determine a new non existant sub directory.
+
+    Features:
+     - With mkdir: Thread and process save.
+     - Different conventions for ID naming possible, default running index.
+     - MPI aware: Get the folder on one worker and distribute to others.
+
+    Args:
+        basedir:
+            The new folder will be inside this directory
+        id_naming:
+            The id naming that is used for the folder name.
+             - str: 'index':
+                The largest index in basedir + 1.
+                e.g.: '1', '2', ...
+             - str: 'time': A timestamp with the format %Y-%m-%d-%H-%M-%S
+                e.g. '2020-08-13-17-02-57'
+             - callable: Each call should generate a new name.
+        mkdir:
+            Creates the dir and makes the program process/thread safe.
+            Note this option ensures that you don't get a
+            conflict between two concurrent calls of get_new_folder.
+            Example:
+                You launch several times your programs and each should get
+                another folder (e.g. hyperparameter search). When inspecting
+                basedir maybe some recognize they can use '2' as sub folder.
+                This option ensures, that only one program gets the '2' and the
+                remaining programs search for another free id.
+        prefix:
+            Optional prefix for the id. e.g.: '2' -> '{prefix}_2'
+        suffix:
+            Optional suffix for the id. e.g.: '2' -> '2_{suffix}'
+        consider_mpi:
+            If True, only search on one mpi process for the folder and
+            distribute the folder name.
+            When using mpi with this option disables the following can/will
```

Comment:
disabled? If disables is correct, I don't get this sentence.

Reply:
I changed the complete statement.

Reply:
I still don't get this sentence

Reply:
Sorry, I changed the text below. Now I updated this line to
`When using mpi (and `consider_mpi is False`) the following can/will happen`